packages/core/src/internal/base/button.base.ts

```diff
@@ -82,17 +82,13 @@ export class CdsBaseButton extends LitElement {
 
   protected firstUpdated(props: Map<string, any>) {
     super.firstUpdated(props);
-    this.updateButtonAttributes();
     this.setupAnchorFocus();
     this.setupNativeButtonBehavior();
   }
 
   protected updated(props: Map<string, any>) {
     super.updated(props);
-    // if readonly or disabled attribute was updated, button attributes might need updating
-    if (props.has('readonly') || props.has('disabled')) {
```

Comment:
This check was preventing the attrs from being updated when inheriting the base button class. Some of the components like tags had role of button even though they were set to readonly. I'm not 100% what was causing this but running the check every time resolves the issue.